File: rhine/src/main/java/com/qingmei2/rhine/http/cache/UserInfoCacheProviders.java
package com.qingmei2.rhine.http.cache;


import com.qingmei2.rhine.http.entity.UserInfo;

import java.util.concurrent.TimeUnit;

import io.reactivex.Maybe;
import io.rx_cache2.DynamicKey;
import io.rx_cache2.EvictDynamicKey;
import io.rx_cache2.LifeCache;


/**
 * Created by QingMei on 2017/8/17.
 * desc:UserInfo缓存
 */

public interface UserInfoCacheProviders {

    @LifeCache(duration = 10, timeUnit = TimeUnit.SECONDS)
    Maybe<UserInfo> getUserInfo(
            Maybe<UserInfo> userInfoObservable,
            DynamicKey dynamicKey,
            EvictDynamicKey evictDynamicKey
    );

}

File: rhine/src/main/java/com/qingmei2/rhine/http/cache/UserInfoCacheProviders.kt
package com.qingmei2.rhine.http.cache

import com.qingmei2.rhine.http.entity.QueryUser
import io.reactivex.Maybe
import io.rx_cache2.DynamicKey
import io.rx_cache2.EvictDynamicKey
import io.rx_cache2.LifeCache
import java.util.concurrent.TimeUnit

interface UserInfoCacheProviders {

    @LifeCache(duration = 10, timeUnit = TimeUnit.SECONDS)
    fun getUserInfo(
            userinfo: Maybe<QueryUser>,
            dynamicKey: DynamicKey,
            evictDynamicKey: EvictDynamicKey
    ): Maybe<QueryUser>
}
